Add audiofile.has_video()

hagenw opened the pull request:
Add `audiofile.has_video()` to check if a given media file contains video.

In `audbcards` we need to inspect media files if they contain video or not. One approach is to use `mediainfo` for this. As `audiofile` relies already on `mediainfo`, I thought it would be a good fit to integrate that function here.

![image](https://github.com/user-attachments/assets/8ee62f35-3c0a-4a5a-8945-9b6ebb88baa2)

schruefer commented:
Hey, thanks for the update! I think it's a great extension.

My only question is how we should handle it if a file does not exist.

If we have an non-existing ["wav", "flac", "mp3", "ogg"] file we get an result
```
>>> has_video('no_file.wav')
False

```
whereas it raises an Error for all other datatypes.
```
>>> has_video('no_file.mp4')
  File "<stdin>", line 1, in <module>
  File "/home/usr/audiofile/audiofile/core/info.py", line 213, in has_video
    video_format = run(cmd)
  File "/home/usr/audiofile/audiofile/core/utils.py", line 86, in run
    out = subprocess.check_output(shell_command, stderr=subprocess.STDOUT)
  File "/usr/lib/python3.8/subprocess.py", line 415, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 516, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['mediainfo', '--Inform=Video;%Format%', 'no_file.mp4']' returned non-zero exit status 1.
```
I am also not getting the `binary_missing_error`, but also "mediainfo cannot be found. Please make sure it is installed" is a little bit none saying to me.

Do you think it makes sense to include an `os.path.isfile(file)` at the beginning?

hagenw commented:
Thanks for testing this out.
I figured out, that we also have a mismatch between the reported error and the documented error for other functions as well and I created https://github.com/audeering/audiofile/issues/154.

For `audiofile.has_video()`, I decided to stay with the current approach for wav, mp3, ogg, flac files, as the functions should be as fast as possible and checking if a file exists takes time. I updated the docstring accordingly:

![image](https://github.com/user-attachments/assets/65c7176f-464c-4cb0-9670-8d5dd449bf41)

and added a test for it.

Now we get:

```python
>>> audiofile.has_video('no_file.wav')
False
>>> audiofile.has_video('no_file.mp4')
...
RuntimeError: 'no_file.mp4' does not exist.
```

Regarding your other comment. When `mediainfo` is missing I get a meaningful error message:

```bash
$ sudo mv /usr/bin/mediainfo /usr/bin/mediainfo.bak
$ touch empty.mp4
$ python -c "import audiofile; audiofile.has_video('empty.mp4')"
...
FileNotFoundError: mediainfo cannot be found.
Please make sure it is installed.
For further instructions visit: https://audeering.github.io/audiofile/installation.html
```

schruefer commented:
Thanks for the updates!

> Regarding your other comment. When mediainfo is missing I get a meaningful error message: 

Yes, this is fine. I meant, that it is no meaningful error message if the file is missing.

Look, good to me.
